staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +181,20 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// The TransformFunc is called for each object which is about to be stored.
+	//
+	// This function is intended for you to take the opportunity to
+	// remove, transform, or normalize fields. One use case is to strip unused
+	// metadata fields out of objects to save on RAM cost.
+	//
+	// Must be set before starting the informer.
+	//
+	// Note: Since the object given to the handler may be already shared with
```

Comment:
How does this happen? Seems like a sharedIndexInformer can only run once, and ProcessDeltas is called in controller.processLoop which is not concurrent.

DeepCopy may be expensive so given one of the primary use cases for this is performance it seems fairly important to not do extra work when not needed?

Reply:
When a relist off existing data occurs it is possible for handledeltas to be called multiple times for the same object. The first time the transformation is applied it will not be concurrent. Assuming you have a way of marking an already- transformed object in order to skip repeated transformation applications then a copy should not be necessary.